packages/server/api/test/unit/ai/ai-mcp-chat.controller.test.ts

```diff
@@ -0,0 +1,332 @@
+    };
+
+    beforeEach(async () => {
+      jest.clearAllMocks();
+
+      handlers = {};
+
+      (getChatContext as jest.Mock).mockResolvedValue(mockChatContext);
+      (getChatHistory as jest.Mock).mockResolvedValue([...mockMessages]);
+      (
+        aiConfigService.getActiveConfigWithApiKey as jest.Mock
+      ).mockResolvedValue(mockAiConfig);
+      (encryptUtils.decryptString as jest.Mock).mockReturnValue(
+        'decrypted-api-key',
+      );
+      (getAiProviderLanguageModel as jest.Mock).mockResolvedValue(
+        mockLanguageModel,
+      );
+      (getMcpSystemPrompt as jest.Mock).mockResolvedValue('system prompt');
+
+      await aiMCPChatController(mockApp, {} as FastifyPluginOptions);
+    });
+
+    it('should call selectRelevantTools with the correct parameters', async () => {
+      (getMCPTools as jest.Mock).mockResolvedValue(mockAllTools);
+
+      const postHandler = handlers['/'];
+      expect(postHandler).toBeDefined();
+
+      await postHandler(
+        mockRequest as FastifyRequest,
+        mockReply as unknown as FastifyReply,
+      );
+
+      expect(selectRelevantTools).toHaveBeenCalledWith({
+        messages: [...mockMessages, { role: 'user', content: 'test message' }],
+        tools: mockAllTools,
+        languageModel: mockLanguageModel,
+        aiConfig: mockAiConfig,
+      });
+    });
+
+    it('should correctly detect presence of analytics and table tools and set isAnalyticsLoaded and isTablesLoaded flags', async () => {
+      (getMCPTools as jest.Mock).mockResolvedValue(mockAllTools);
+      (selectRelevantTools as jest.Mock).mockResolvedValue({
+        tool1: { description: 'Tool 1', parameters: {} },
+        mcp_analytics_superset: {
+          description: 'Analytics tool',
+          parameters: {},
+        },
+        mcp_table_tool: { description: 'Table tool', parameters: {} },
+      });
+
+      const postHandler = handlers['/'];
+      await postHandler(
+        mockRequest as FastifyRequest,
+        mockReply as unknown as FastifyReply,
+      );
+
+      expect(getMcpSystemPrompt).toHaveBeenCalledWith({
+        isAnalyticsLoaded: true,
+        isTablesLoaded: true,
+      });
+    });
+
+    it('should correctly detect absence of analytics and table tools and set isAnalyticsLoaded and isTablesLoaded flags', async () => {
+      (getMCPTools as jest.Mock).mockResolvedValue(mockNoSpecialTools);
+      (selectRelevantTools as jest.Mock).mockResolvedValue({
+        tool1: { description: 'Tool 1', parameters: {} },
+        tool2: { description: 'Tool 2', parameters: {} },
+      });
+
+      const postHandler = handlers['/'];
+      await postHandler(
+        mockRequest as FastifyRequest,
+        mockReply as unknown as FastifyReply,
+      );
+
+      expect(getMcpSystemPrompt).toHaveBeenCalledWith({
+        isAnalyticsLoaded: false,
+        isTablesLoaded: false,
+      });
+    });
+
+    it('should handle when selectRelevantTools returns undefined', async () => {
```

Comment:
you can reuse test code with 
`test.each([undefined, {}, null], async (selectedTools) =>`